openshift-201/pipelines.md

```diff
@@ -0,0 +1,232 @@
+
+</br>
+
+## Pipelines
+A `Pipeline` is a collection of `Task` resources arranged in a specific order of execution. They are executed to construct complex workflows that automate the build, deployment and delivery of applications. You can define a CI/CD workflow for your application using pipelines containing one or more tasks.
+
+The following example shows the [`mvn-build`](https://github.com/bcgov/pipeline-templates/blob/main/tekton/base/pipelines/maven.yaml) pipeline.
+
+
+![pipeline](images/pipelines/pipeline-yaml.png)
+
+
+1. Defines the Pipeline API version and type of Kubernetes resource
+2. Defines the parameters (inputs) used in this Pipeline
+3. Defines the workspaces (shared storage volumes) used by the Pipeline
+4. Defines the tasks in the Pipeline.
+    - `taskRef`: Refers to the task to run
+    - `runAfter`: Allows for ordering of tasks (otherwise, the default is run in parallel)
+    - `params`: inputs to the task
+    - `workspaces`: shared storage volumes for the task
+
+
+</br>
+
+The details of the pipeline are as follows:
+![pipeline details](images/pipelines/pipeline-details.png)
+
+Notice the `mvn-run` and `mvn-sonar-scan` tasks are run in parallel.  The triangle before the `mvn-sonar-scan` indicates a `when` condition as shown here:
+```yaml
+ - name: mvn-sonar-scan
+    when:
+    - input: "$(params.runSonarScan)"
+        operator: in
+        values: ["true"]
+```
+This is checking the `runSonarScan` parameter defined in the `Pipeline` and if `true` will run this `Task`.  Otherwise, it will be skipped.
+
+## Running a Pipeline
+Pipelines can be triggered either manually by creating a `PipelineRun` or automatically via a `Trigger`.  Let's manually create a `PipelineRun`.  This can be done via the command line by running:
+```bash
+cat <<EOF | kubectl create -f -
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: mvn-build-run-
+spec:
+  pipelineRef:
+    name: p-mvn-build
+  params:
+  - name: appName
+    value: maven-test
+  - name: repoUrl
+    value: git@github.com:bcgov/pipeline-templates.git
+  - name: branchName
+    value: main
+  - name: pathToContext
+    value: ./tekton/demo/maven-test
+  - name: mavenImage
+    value: index.docker.io/library/maven
+  - name: runSonarScan
+    value: 'true'
+  - name: sonarProject
+    value: bcgov_pipeline-templates
+  workspaces:
+  - name: shared-data
+    volumeClaimTemplate:
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
+  - name: ssh-creds
+    secret:
+      secretName: ssh-key-path
+  - name: maven-settings
+    emptyDir: {}
+EOF
+```
+or via the OpenShift Console by performing the following:
+
+![pipeline console 01](images/pipelines/pipeline-console-01.png)
+
+After you've selected the `p-mvn-build` then do the following:
```

Comment:
Each of these should have a blank line following to ensure the image is displayed on its own line and not at the end of the line of the description
```suggestion
After you've selected the `p-mvn-build` then do the following:

```

Reply:
There were other images that needed a blank line as well.  Addressed them all